lib/screens/settings/settings_screen.dart

```diff
@@ -83,44 +82,108 @@ class SettingsScreen extends StatelessWidget {
+          void _togglePasswordVisibility() {
+            setState(() {
+              _passwordVisible = !_passwordVisible;
+            });
+          }
+
+          return AlertDialog(
+            title: Text("Change password"),
+            content: Form(
+              key: _formKey,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  TextFormField(
+                    controller: _currentPassController,
+                    decoration: InputDecoration(
+                      labelText: "Current password",
+                      suffixIcon: IconButton(
+                        icon: Icon(_passwordVisible ? Icons.visibility : Icons.visibility_off),
+                        onPressed: _togglePasswordVisibility,
+                      ),
+                    ),
+                    validator: (value) => _validatePasswords(value),
+                    obscureText: !_passwordVisible,
+                  ),
+                  TextFormField(
+                    controller: _newPassController,
+                    decoration: InputDecoration(
+                      labelText: "New password",
+                      suffixIcon: IconButton(
+                        icon: Icon(_passwordVisible ? Icons.visibility : Icons.visibility_off),
+                        onPressed: _togglePasswordVisibility,
+                      ),
+                    ),
+                    validator: (value) => _validatePasswords(value),
+                    obscureText: !_passwordVisible,
+                  ),
+                  TextFormField(
+                    controller: _newPassConfirmController,
+                    decoration: InputDecoration(
+                      labelText: "Confirm password",
+                      suffixIcon: IconButton(
+                        icon: Icon(_passwordVisible ? Icons.visibility : Icons.visibility_off),
+                        onPressed: _togglePasswordVisibility,
+                      ),
+                    ),
+                    obscureText: !_passwordVisible,
+                    validator: (value) => _validateConfirmPass(value),
+                  ),
+                ],
+              ),
             ),
-          ],
-        ),
-        actions: [
-          FlatButton(
-            child: Text("Submit"),
-            onPressed: () async {
-              ChangePassword changePassword = ChangePassword(
-                currentPassword: _currentPassController.text,
-                newPassword: _newPassController.text,
-              );
-              Navigator.of(context).pop();
-              showProgressIndicator(context);
-              try {
-                CustomResponse response =
-                    await UserRepository.instance.changePassword(changePassword);
-                topContext.showSnackBar(response.message);
-              } on Failure catch (failure) {
-                topContext.showSnackBar(failure.message);
```

Comment:
@bharat-1809 did you test the apk? iirc topContext is needed

Reply:
@Techno-Disaster sorry my bad. I have now checked and resolved the issue and its working fine. Please have a look :)
Thanks